muxers/yamux/src/lib.rs

```diff
@@ -179,16 +179,23 @@ where
 
     fn close(&self, c: &mut Context<'_>) -> Poll<YamuxResult<()>> {
         let mut inner = self.0.lock();
-        if let std::task::Poll::Ready(x) = Pin::new(&mut inner.control).poll_close(c) {
-            return Poll::Ready(x.map_err(YamuxError));
+
+        if let Poll::Ready(()) = Pin::new(&mut inner.control)
+            .poll_close(c)
+            .map_err(YamuxError)?
+        {
+            return Poll::Ready(Ok(()));
         }
-        while let std::task::Poll::Ready(x) = inner.incoming.poll_next_unpin(c) {
-            match x {
-                Some(Ok(_)) => {} // drop inbound stream
-                Some(Err(e)) => return Poll::Ready(Err(e)),
+
+        while let Poll::Ready(_inbound_stream) =
```

Comment:
```suggestion
        while let Poll::Ready(inbound_stream) =
```
Why prefix `inbound_stream` with an `_`?

Reply:
To make it more obvious that it will be unused but I guess _technically_ it is not unused!

Reply:
I've tried to improve this with [6eb64fe](https://github.com/libp2p/rust-libp2p/pull/2677/commits/6eb64fe8155a6a18f7d67333fb9c43c3a50885cc).